### PR TITLE
Reduce Trigger Descriptor map size

### PR DESCRIPTION
**Pull request by DavidTanner**
Many of the variables used in GhprbPullRequest can be grabbed as desired instead of stored as a permanent part of the object.  Some of them can also be fetched on init from the repo and some only matter for a single check.

### CODE REVIEW

**Comment by DavidTanner** [comment]
@arcivanov please review?

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 1]
```diff
@@ -27,31 +26,35 @@
 public class GhprbPullRequest {
 
     private static final Logger logger = Logger.getLogger(GhprbPullRequest.class.getName());
+    
+    @Deprecated @SuppressWarnings("unused") private transient GHUser author;
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Do we have to keep these? What is the benefit?

Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: https://wiki.jenkins-ci.org/display/JENKINS/Hint+on+retaining+backward+compatibility

Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: If the fields are removed then Jenkins will toss all of the old entries and build every PR again.

Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: understood

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 2]
```diff
@@ -396,7 +383,7 @@ public int hashCode() {
         hash = 89 * hash + this.id;
         return hash;
     }
-
+
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: formatting

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 3]
```diff
@@ -435,15 +427,15 @@ public String getTitle() {
      * @return the Github Pull Request URL
      */
     public URL getUrl() {
-        return url;
+        return pr.getHtmlUrl();
     }
-
+
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: formatting

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 4]
```diff
@@ -422,11 +409,16 @@ public String getSource() {
     }
 
     public String getAuthorEmail() {
-        return authorEmail;
+        try {
+            return pr.getUser().getEmail();
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Unable to fetch author info for " + id);
```
Comment (arcivanov) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Is it a concern that the number of round trips to GH will cause rate limit failures?

**Comment by arcivanov** [comment]
this has been merged to #202

**Pull request closed without merging**